Simplenote/src/main/res/values/colors.xml

```diff
@@ -235,4 +235,7 @@
     <color name="yellow_90">#332200</color>
     <color name="yellow_100">#1c1300</color>
 
+    <!-- Update the status guard color to avoid a black status bar on ActionMode -->
+    <color name="abc_input_method_navigation_guard" tools:override="true">@android:color/transparent</color>
```

Comment:
Using the postDelayed to restore the transparent status bar color when action mode was destroyed was causing some flickering after my changes, as I had to run `setActivateOnItemClick` before refreshing the list to avoid resetting the selection.
Overriding this color fixes the issue without having to change the status bar color.

Reply:
I'm not a fan of overriding third-party library resources or using reflection since we don't have complete control over those resources.  Can we avoid overriding this resource if we leave the `postDelayed` code in `onDestroyActionMode`, only update the status bar color in that  `Runnable`, move the `setActivateOnItemClick` statement with the associated code outside the `Runnable`, and add the status bar setting statement back to `onCreateActionMode`?  I think we can replace `getActivity()` with `requireActivity()` too.  It would look something like this.

```java
@Override
public boolean onCreateActionMode(ActionMode actionMode, Menu menu) {
    mCallbacks.onActionModeCreated();
    MenuInflater inflater = actionMode.getMenuInflater();
    inflater.inflate(R.menu.bulk_edit, menu);
    DrawableUtils.tintMenuWithAttribute(getActivity(), menu, R.attr.actionModeTextColor);
    mActionMode = actionMode;
    requireActivity().getWindow().setStatusBarColor(ThemeUtils.getColorFromAttribute(requireContext(), R.attr.mainBackgroundColor));
    return true;
}
```

```java
@Override
public void onDestroyActionMode(ActionMode mode) {
    mCallbacks.onActionModeDestroyed();
    mActionMode = null;
    new Handler().postDelayed(
        new Runnable() {
            @Override
            public void run() {
                requireActivity().getWindow().setStatusBarColor(getResources().getColor(android.R.color.transparent, requireActivity().getTheme()));
            }
        },
        requireContext().getResources().getInteger(android.R.integer.config_mediumAnimTime)
    );

    NotesActivity notesActivity = (NotesActivity) requireActivity();
    setActivateOnItemClick(DisplayUtils.isLargeScreenLandscape(notesActivity));

    if (mSelectedNoteId == null) {
        notesActivity.showDetailPlaceholder();
    }
}
```

Reply:
This was the first thing I tried, but as `setActivateOnItemClick` will cause clearing the selection, I noticed that there is a blink of the black status bar.
But now when I think about it, the status guard is added by the compat library only if the status bar is transparent, and we don't need transparent status bars in the app I think, I changed the statusBarColor to match the background color in this commit: 1d264aec5b255deaabf586ea2ef4f82c2b7654f5, which solves the issue without impacting the UI, unless I'm missing something, or an edge case.

Reply:
That was the first thing that I tried, but I noticed some blinking in the status bar, personally I prefer the color override, as both solutions are kind of hackish, the fact that we need to post reverting to the transparent color depends on the internals of the library too, and could break with a library update 🙂, I wish they offered a way to disable this behavior for transparent status bar..
I was able to solve the blinking issue by using a longer duration for the delay: f4ff27ffc4eeb5c1d74f3c7cfc403638a3a75823